server/server.go

```diff
@@ -30,6 +31,7 @@ type Server struct {
 	Config               *Config
 	Store                store.Store
 	GithubClient         *GithubClient
+	CircleCiClient       *circleci.Client
```

Comment:
Late review: It would be great if `circleci.Client` would become an interface similar to `GithubClient`. That would allow proper unit tests for methods that use CircleCi. Should I open a JIRA ticket or GitHub issue for that idea?

Reply:
Yes please ;)

Reply:
Totally agree. Please file it under Mattermod revamp epic and under Platform team. Thanks!

Reply:
Done: https://mattermost.atlassian.net/browse/MM-26704